trailblazer/services/slurm/slurm_api_service/utils.py

```diff
@@ -0,0 +1,34 @@
+from datetime import datetime
+from trailblazer.clients.slurm_api_client.dto.common import SlurmAPIJobInfo
+from trailblazer.clients.slurm_api_client.dto.job_response import SlurmJobResponse
+from trailblazer.constants import TrailblazerStatus
+from trailblazer.services.slurm.dtos import SlurmJobInfo
+
+
+def create_job_info_dto(job_response: SlurmJobResponse) -> SlurmJobInfo:
+    job: SlurmAPIJobInfo = job_response.jobs[0]
+    elapsed: int | None = get_job_elapsed_time(job)
+    start_time: datetime | None = get_job_start_time(job)
+    status: TrailblazerStatus | None = get_job_state(job)
+
+    return SlurmJobInfo(
+        slurm_id=job.job_id,
+        status=status,
+        name=job.name,
+        started_at=start_time,
+        elapsed=elapsed,
+    )
+
+
+def get_job_elapsed_time(job: SlurmAPIJobInfo) -> int:
+    return job.accrue_time.number if job.accrue_time and job.accrue_time.set else 0
+
+
+def get_job_start_time(job: SlurmAPIJobInfo) -> datetime | None:
+    if job.start_time is None or job.start_time.set is None:
```

Comment:
What do the `.set` fields do?

Reply:
That is a field provided by the slurm API in the response to indicate whether the field has been set. I think it is redundant and I'll remove it

Reply:
Removed in https://github.com/Clinical-Genomics/trailblazer/pull/451/commits/81b0af1a637b1123def6fa402758fbdc4c52e72e